.github/workflows/prerelease.yml

```diff
@@ -91,158 +91,161 @@ jobs:
-      - name: Upload Release Asset TAR
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ needs.create_prerelease.outputs.UPLOAD_URL }}
-          asset_path: ./exporters/${{ steps.vars.outputs.NAME }}/target/packages/${{ steps.vars.outputs.PACKAGE_NAME }}_linux_${{ steps.vars.outputs.VERSION }}_amd64.tar.gz
-          asset_name: ${{ steps.vars.outputs.PACKAGE_NAME }}_linux_${{ steps.vars.outputs.VERSION }}_amd64.tar.gz
-          asset_content_type: tarball
-      - name: Upload Release Asset RPM
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Upload assets to be released
+        uses: actions/upload-artifact@v3
         with:
-          upload_url: ${{ needs.create_prerelease.outputs.UPLOAD_URL }}
-          asset_path: ./exporters/${{ steps.vars.outputs.NAME }}/target/packages/${{ steps.vars.outputs.PACKAGE_NAME }}-${{ steps.vars.outputs.VERSION }}-1.x86_64.rpm
-          asset_name: ${{ steps.vars.outputs.PACKAGE_NAME }}-${{ steps.vars.outputs.VERSION }}-1.x86_64.rpm
-          asset_content_type: rpm package
+          name: linux-artifacts
+          retention-days: 1
+          path: |
+            ./exporters/${{ steps.vars.outputs.NAME }}/target/packages/*_${{ steps.vars.outputs.VERSION }}-1_*.deb
+            ./exporters/${{ steps.vars.outputs.NAME }}/target/packages/*_linux_${{ steps.vars.outputs.VERSION }}_*.tar.gz
+            ./exporters/${{ steps.vars.outputs.NAME }}/target/packages/*-${{ steps.vars.outputs.VERSION }}-1.*.rpm
 
-
-  ###
-  ###   Windows Packaging # In case a new release has been created and Windows packages are needed build and push MSI
-  ###
-  build_push_windows_artifacts:
-    name: Create MSI
+  build_windows_artifacts:
+    name: Build Windows artifacts and packages
     runs-on: windows-latest
-    needs: [create_prerelease]
-    if: ${{ needs.create_prerelease.outputs.CREATE_RELEASE == 'true' && needs.create_prerelease.outputs.PACKAGE_WINDOWS == 'true' }}
+    needs: [check_exporter_preconditions]
+    if: ${{ needs.check_exporter_preconditions.outputs.CREATE_RELEASE == 'true' && needs.check_exporter_preconditions.outputs.PACKAGE_WINDOWS == 'true' }}
     env:
       GOPATH: ${{ github.workspace }}
       GOOS: windows
     defaults:
       run:
         working-directory: src/repo
-    strategy:
-      matrix:
-        goarch: [amd64]
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           path: src/repo
-      - uses: actions/setup-go@v2
+          fetch-depth: 0
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: src/repo/nri-config-generator/go.mod
+      - name: Install goreleaser
+        uses: goreleaser/goreleaser-action@v4
         with:
-          go-version: ${{env.GO_VERSION}}
+          version: v1.13.1
```

Comment:
should we replace this as well?

Reply:
Sure, I missed that one and also missed this comment before merging, I'll replace it in a follow-up PR. Thanks!